engine/lite/components/kafka/src/test/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaAvroFunctionalTest.scala

```diff
@@ -0,0 +1,68 @@
+package pl.touk.nussknacker.engine.lite.components
+
+import org.scalatest.{FunSuite, Matchers}
+import pl.touk.nussknacker.engine.api.process.ProcessObjectDependencies
+import pl.touk.nussknacker.engine.avro.AvroUtils
+import pl.touk.nussknacker.engine.avro.encode.ValidationMode
+import pl.touk.nussknacker.engine.avro.schemaregistry.SchemaVersionOption
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.client.{MockConfluentSchemaRegistryClientFactory, MockSchemaRegistryClient}
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.lite.util.test.{KafkaConsumerRecord, LiteKafkaTestScenarioRunner}
+import pl.touk.nussknacker.engine.util.namespaces.DefaultNamespacedObjectNaming
+
+class LiteKafkaAvroFunctionalTest extends FunSuite with Matchers {
+
+  import LiteKafkaComponentProvider._
+  import LiteKafkaTestScenarioRunner._
+  import pl.touk.nussknacker.engine.avro.KafkaAvroBaseComponentTransformer._
+  import pl.touk.nussknacker.engine.spel.Implicits._
+
+  private val recordSchema = AvroUtils.parseSchema(
+    """{
+      |  "type": "record",
+      |  "namespace": "pl.touk.nussknacker.engine.avro",
+      |  "name": "FullName",
+      |  "fields": [
+      |    { "name": "first", "type": "string" },
+      |    { "name": "last", "type": "string" },
+      |    { "name": "age", "type": "int" },
+      |    { "name": "address", "type": {"type":"record", "name": "Address", "fields": [{ "name": "city", "type": "string" }]}}
+      |  ]
+      |}
+    """.stripMargin)
+
+  private val inputTopic = "input"
+  private val outputTopic = "output"
+
+
+  test("should test end to end kafka avro sink / source") {
+    //Given
+    val scenario = ScenarioBuilder.streamingLite("check avro serialization")
+      .source("my-source", KafkaAvroName, TopicParamName -> s"'$inputTopic'", SchemaVersionParamName -> s"'${SchemaVersionOption.LatestOptionName}'")
+      .emptySink("my-sink", KafkaSinkRawAvroName, TopicParamName -> s"'$outputTopic'",  SchemaVersionParamName -> s"'${SchemaVersionOption.LatestOptionName}'", SinkKeyParamName -> "", SinkValueParamName -> s"#input", SinkValidationModeParameterName -> s"'${ValidationMode.strict.name}'")
+
+    val (mockedComponents, mockSchemaRegistryClient) = prepareRunnerConfig
+    val runtime = LiteKafkaTestScenarioRunner(mockSchemaRegistryClient, mockedComponents)
+    runtime.registerSchemaAvro(inputTopic, recordSchema)
+    runtime.registerSchemaAvro(outputTopic, recordSchema)
+
+    //When
+    val record = AvroUtils.createRecord(recordSchema,
```

Comment:
Can u use LogicalTypesGenericRecordBuilder to reduce "magic" in generation of input data?

Reply:
If you want to use `LogicalTypesGenericRecordBuilder` you still can do it. If someone wants se how is implemented creation record then he can go inside method. For me it's more convenient way to create GenericRecord.

Reply:
Doing such things you can have false positives in tests. Verification code should be as stupid as it is possible - your verification code has magic of conversion between map and generic record.

Reply:
Test will not pass because record will be not compatibility with schema. But ok, I will change it. But to be honest with you, I I would rather write test an use magic conversion then use building record step by step. The same think I can say about: 
`ConfluentUtils.serializeRecordToBytesArray` what if confluent will change their serialization way?

Reply:
> record will be not compatibility with schema
Will be if field will be optional - it is one of scenarios that u will assume in tests. Won't you?
> what if confluent will change their serialization way
I don't get this comparison

To be honest, at the beginning I was thinking that you are using `BestEffortAvroEncoder` for that, but after that I've realized that you wrote almost no-magic map to record translation. It is ok for me - I was scary about  `BestEffortAvroEncoder` magic

Reply:
Ah.. Okey, so I will rollback it! :)